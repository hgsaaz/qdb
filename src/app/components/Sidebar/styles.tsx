import { Layout, Col } from 'antd';
import styled from 'styled-components/macro';
const { Sider } = Layout;

export const SiderWrapper = styled(Sider)`
  padding: 30px;
`;

export const CenterCol = styled(Col)`
  text-align: center;
`;

export const UserInfo = styled.p`
  padding: 0;
  margin: 0;
  padding-bottom: 10px;
  font-weight: bold;
`;

export const WelcomeText = styled.p`
  padding: 0;
  margin: 0;
  padding-top: 10pt;
`;
