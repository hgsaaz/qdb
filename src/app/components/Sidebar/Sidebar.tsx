import React from 'react';
import { Avatar, Row, Col, Button, Space, Divider } from 'antd';
import 'antd/dist/antd.less';
import { Menu } from 'antd';
import { ThunderboltOutlined } from '@ant-design/icons';
import { SiderWrapper, CenterCol, UserInfo, WelcomeText } from './styles';
import { useHistory } from 'react-router-dom';

function Sidebar(props) {
  let histroy = useHistory();
  const handleClick = item => {
    let path = '/';
    if (item.key > 2) {
      path = '/reports';
    } else {
      path = '/';
    }

    if (histroy.location.pathname !== path) {
      histroy.push(path);
    }
  };

  return (
    <SiderWrapper>
      <Row gutter={[16, 16]} justify="center">
        <CenterCol span={24}>
          <Avatar
            size={90}
            src="https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png"
          />

          <Space size="small" direction="vertical">
            <WelcomeText>Hello</WelcomeText>
            <UserInfo>{props.user && props.user.name}</UserInfo>
            <Button type="primary" icon={<ThunderboltOutlined />}>
              Live Metrices
            </Button>
          </Space>
          <Divider />
        </CenterCol>
        <Col span={24}>
          <Menu defaultSelectedKeys={['1']} mode="inline" onClick={handleClick}>
            <Menu.ItemGroup key="g1" title="Dashboard">
              <Menu.Item key="1">Overview</Menu.Item>
              <Menu.Item key="2">Expense</Menu.Item>
            </Menu.ItemGroup>
            <Menu.ItemGroup key="g2" title="Reports">
              <Menu.Item key="3">Active Reports</Menu.Item>
              <Menu.Item key="4">Calender</Menu.Item>
              <Menu.Item key="5">Live Stats</Menu.Item>
            </Menu.ItemGroup>
          </Menu>
        </Col>
      </Row>
    </SiderWrapper>
  );
}

export default Sidebar;
