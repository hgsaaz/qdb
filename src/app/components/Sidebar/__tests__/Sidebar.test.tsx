import React from 'react';
import { Router } from 'react-router-dom';
import { render, act } from '@testing-library/react';
import Sidebar from '../Sidebar';
import { createMemoryHistory } from 'history';
const history = createMemoryHistory();
const renderSidebar = (props: Parameters<typeof Sidebar>[number] = {}) =>
  render(
    <Router history={history}>
      <Sidebar {...props} />
    </Router>,
  );
const name = 'testUser';
const testuser = {
  name: name,
};

describe('<Sidebar />', () => {
  beforeAll(() => {
    // mocking matchMedia
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match the snapshot', () => {
    const { container } = renderSidebar({
      user: testuser,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have props displayed', () => {
    act(() => {
      const { queryByText } = renderSidebar({
        user: testuser,
      });
      expect(queryByText(name)).toBeInTheDocument();
    });
  });
});
