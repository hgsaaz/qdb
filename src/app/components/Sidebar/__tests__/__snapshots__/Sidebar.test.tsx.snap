// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match the snapshot 1`] = `
.c0 {
  padding: 30px;
}

.c1 {
  text-align: center;
}

.c3 {
  padding: 0;
  margin: 0;
  padding-bottom: 10px;
  font-weight: bold;
}

.c2 {
  padding: 0;
  margin: 0;
  padding-top: 10pt;
}

<aside
  class="c0 ant-layout-sider ant-layout-sider-dark"
  style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
>
  <div
    class="ant-layout-sider-children"
  >
    <div
      class="ant-row ant-row-center"
      style="margin: -8px -8px 8px -8px;"
    >
      <div
        class="ant-col ant-col-24 c1"
        style="padding: 8px 8px 8px 8px;"
      >
        <span
          class="ant-avatar ant-avatar-circle ant-avatar-image"
          style="width: 90px; height: 90px; line-height: 90px; font-size: 18px;"
        >
          <img
            src="https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png"
          />
        </span>
        <div
          class="ant-space ant-space-vertical"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <p
              class="c2"
            >
              Hello
            </p>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <p
              class="c3"
            >
              testUser
            </p>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span
                aria-label="thunderbolt"
                class="anticon anticon-thunderbolt"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="thunderbolt"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M848 359.3H627.7L825.8 109c4.1-5.3.4-13-6.3-13H436c-2.8 0-5.5 1.5-6.9 4L170 547.5c-3.1 5.3.7 12 6.9 12h174.4l-89.4 357.6c-1.9 7.8 7.5 13.3 13.3 7.7L853.5 373c5.2-4.9 1.7-13.7-5.5-13.7zM378.2 732.5l60.3-241H281.1l189.6-327.4h224.6L487 427.4h211L378.2 732.5z"
                  />
                </svg>
              </span>
              <span>
                Live Metrices
              </span>
            </button>
          </div>
        </div>
        <div
          class="ant-divider ant-divider-horizontal"
          role="separator"
        />
      </div>
      <div
        class="ant-col ant-col-24"
        style="padding: 8px 8px 8px 8px;"
      >
        <ul
          class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
          role="menu"
        >
          <li
            class=" ant-menu-item-group"
          >
            <div
              class="ant-menu-item-group-title"
              title="Dashboard"
            >
              Dashboard
            </div>
            <ul
              class="ant-menu-item-group-list"
            >
              <li
                class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Overview
              </li>
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Expense
              </li>
            </ul>
          </li>
          <li
            class=" ant-menu-item-group"
          >
            <div
              class="ant-menu-item-group-title"
              title="Reports"
            >
              Reports
            </div>
            <ul
              class="ant-menu-item-group-list"
            >
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Active Reports
              </li>
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Calender
              </li>
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Live Stats
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>
</aside>
`;
